control-plane/agents/core/src/pool/states.rs

```diff
@@ -0,0 +1,26 @@
+use crate::core::states::ResourceStates;
+use common_lib::types::v0::store::{pool::PoolState, replica::ReplicaState};
+
+/// Implementation of the ResourceSpecs which is retrieved from the ResourceSpecsLocked
+/// During these calls, no other thread can add/remove elements from the list
+impl ResourceStates {
+    /// Get all ReplicaStates
+    pub(crate) async fn get_replicas(&self) -> Vec<ReplicaState> {
```

Comment:
Also generic.. 

impl RefreshState for ... ?

Reply:
I have created a generic function to get a vector of states but haven't created a trait yet. The next set of changes I am planning to make is to change the way the NodeWrapper is used. So I would like to leave the trait part of this for now and tackle that in my next PR.